pyiron_contrib/atomistics/atomistics/job/structurestorage.py

```diff
@@ -442,60 +472,241 @@ def to_hdf(self, hdf, group_name="structures"):
-                self._per_structure_arrays["cell"] = hdf_s_lst["cells"]
+    >>> container.get_structure(frame=0) == container.get_structure(frame="fcc")
+    True
+
+    Custom arrays may also be defined on the container
+
+    >>> container.add_array("energy", shape=(), dtype=np.float64, fill=-1, per="chunk")
 
-                self._per_structure_arrays["pbc"] = np.full((self.num_structures, 3), True)
+    (chunk means structure in this case, see below and :class:`.FlattenedStorage`)
+
+    You can then pass arrays of the corresponding shape to :method:`add_structure()`
+
+    >>> container.add_structure(Atoms(...), "grain_boundary", energy=3.14)
+
+    Saved arrays are accessed with :method:`.get_array()`
+
+    >>> container.get_array("energy", 3)
+    3.14
+    >>> container.get_array("energy", 0)
+    -1
+
+    It is also possible to use the same names in :method:`.get_array()` as in :method:`.get_structure()`.
+
+    >>> container.get_array("energy", 0) == container.get_array("energy", "fcc")
+    True
+
+    The length of the container is the number of structures inside it.
+
+    >>> len(container)
+    4
+
+    Each structure corresponds to a chunk in :class:`.FlattenedStorage` and each atom to an element.  By default the
+    following arrays are defined for each structure:
+        - identifier    shape=(),    dtype=str,          per chunk; human readable name of the structure
+        - cell          shape=(3,3), dtype=np.float64,   per chunk; cell shape
+        - pbc           shape=(3,),  dtype=bool          per chunk; periodic boundary conditions
+        - symbols:      shape=(),    dtype=str,          per element; chemical symbol
+        - positions:    shape=(3,),  dtype=np.float64,   per element: atomic positions
+    If a structure has spins/magnetic moments defined on its atoms these will be saved in a per atom array as well.  In
+    that case, however all structures in the container must either have all collinear spins or all non-collinear spins.
+    """
+
+    def __init__(self, num_atoms=1, num_structures=1):
+        """
+        Create new structure container.
+
+        Args:
+            num_atoms (int): total number of atoms across all structures to pre-allocate
+            num_structures (int): number of structures to pre-allocate
+        """
+        super().__init__(num_elements=num_atoms, num_chunks=num_structures)
+
+    def _init_arrays(self):
+        super()._init_arrays()
+        # 2 character unicode array for chemical symbols
+        self._per_element_arrays["symbols"] = np.full(self._num_elements_alloc, "XX", dtype=np.dtype("U2"))
+        self._per_element_arrays["positions"] = np.empty((self._num_elements_alloc, 3))
+
+        self._per_chunk_arrays["cell"] = np.empty((self._num_chunks_alloc, 3, 3))
+        self._per_chunk_arrays["pbc"] = np.empty((self._num_elements_alloc, 3), dtype=bool)
+
+
+    @property
+    def symbols(self):
+        """:meta private:"""
```

Comment:
`:meta private:` marks the attributes as private for sphinx, so they don't end up in the docs.

My vision of the API was to use only `set_array`/`get_array` for accessing the data, since these give you the values per chunk, instead of allowing to mess with the bare arrays.  These getters only exist for compatibility with the atomicrex code where the `StructureStorage` was introduced, so I initially didn't want to support them and therefore hide them.  If everyone else agrees that they are useful to have I can properly document them though.